Adapter tests failing with Jakarta error

### Before reporting an issue

- [X] I have searched existing issues
- [X] I have reproduced the issue with the [latest nightly release](https://github.com/keycloak/keycloak/releases/tag/nightly)

### Area

adapter/spring

### Describe the bug

Springboot failing when running it with undertow giving this error

[INFO] org.springframework.context.ApplicationContextException: Unable to start web server; nested exception is java.lang.NoClassDefFoundError: jakarta/servlet/ServletException
[INFO]  at org.springframework.boot.web.servlet.context.ServletWebServerApplicationContext.onRefresh(ServletWebServerApplicationContext.java:162) ~[spring-boot-2.4.13.jar:2.4.13]
[INFO]  at org.springframework.context.support.AbstractApplicationContext.refresh(AbstractApplicationContext.java:577) ~[spring-context-5.3.13.jar:5.3.13]
[INFO]  at org.springframework.boot.web.servlet.context.ServletWebServerApplicationContext.refresh(ServletWebServerApplicationContext.java:144) ~[spring-boot-2.4.13.jar:2.4.13]
[INFO]  at org.springframework.boot.SpringApplication.refresh(SpringApplication.java:771) ~[spring-boot-2.4.13.jar:2.4.13]
[INFO]  at org.springframework.boot.SpringApplication.refresh(SpringApplication.java:763) ~[spring-boot-2.4.13.jar:2.4.13]
[INFO]  at org.springframework.boot.SpringApplication.refreshContext(SpringApplication.java:438) ~[spring-boot-2.4.13.jar:2.4.13]
[INFO]  at org.springframework.boot.SpringApplication.run(SpringApplication.java:339) ~[spring-boot-2.4.13.jar:2.4.13]
[INFO]  at org.springframework.boot.SpringApplication.run(SpringApplication.java:1329) ~[spring-boot-2.4.13.jar:2.4.13]
[INFO]  at org.springframework.boot.SpringApplication.run(SpringApplication.java:1318) ~[spring-boot-2.4.13.jar:2.4.13]
[INFO]  at org.keycloak.SpringBootAdapterApplication.main(SpringBootAdapterApplication.java:10) ~[classes/:na]
[INFO] Caused by: java.lang.NoClassDefFoundError: jakarta/servlet/ServletException
[INFO]  at io.undertow.servlet.core.ServletContainerImpl.addDeployment(ServletContainerImpl.java:51) ~[undertow-servlet-2.3.2.Final.jar:2.3.2.Final]
[INFO]  at org.springframework.boot.web.embedded.undertow.UndertowServletWebServerFactory.createManager(UndertowServletWebServerFactory.java:346) ~[spring-boot-2.4.13.jar:2.4.13]
[INFO]  at org.springframework.boot.web.embedded.undertow.UndertowServletWebServerF
[INFO] actory.getWebServer(UndertowServletWebServerFactory.java:316) ~[spring-boot-2.4.13.jar:2.4.13]
[INFO]  at org.springframework.boot.web.servlet.context.ServletWebServerApplicationContext.createWebServer(ServletWebServerApplicationContext.java:181) ~[spring-boot-2.4.13.jar:2.4.13]
[INFO]  at org.springframework.boot.web.servlet.context.ServletWebServerApplicationContext.onRefresh(ServletWebServerApplicationContext.java:159) ~[spring-boot-2.4.13.jar:2.4.13]
[INFO]  ... 9 common frames omitted
[INFO] Caused by: java.lang.ClassNotFoundException: jakarta.servlet.ServletException
[INFO]  at java.base/jdk.internal.loader.BuiltinClassLoader.loadClass(BuiltinClassLoader.java:641) ~[na:na]
[INFO]  at java.base/jdk.internal.loader.ClassLoaders$AppClassLoader.loadClass(ClassLoaders.java:188) ~[na:na]
[INFO]  at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:525) ~[na:na]
[INFO]  ... 14 common frames omitted

### Version

999-SNAPSHOT

### Expected behavior

Pass

### Actual behavior

Fail

### How to Reproduce?

Run the adapter tests with undertow

### Anything else?

_No response_

--- a/pom.xml
+++ b/pom.xml
@@ -115,7 +115,8 @@
         <sun.xml.ws.version>4.0.0</sun.xml.ws.version>
         <sun.activation.version>1.2.2</sun.activation.version>
         <org.glassfish.jaxb.xsom.version>2.3.3-b02</org.glassfish.jaxb.xsom.version>
-        <undertow.version>2.2.24.Final</undertow.version>
+        <undertow.version>${undertow-legacy.version}</undertow.version>
+        <undertow-legacy.version>2.2.24.Final</undertow-legacy.version>
         <undertow-jakarta.version>2.3.2.Final</undertow-jakarta.version>
         <wildfly-elytron.version>2.2.0.Final</wildfly-elytron.version>
         <elytron.undertow-server.version>1.9.0.Final</elytron.undertow-server.version>

--- a/testsuite/integration-arquillian/test-apps/spring-boot-adapter-app/pom.xml
+++ b/testsuite/integration-arquillian/test-apps/spring-boot-adapter-app/pom.xml
@@ -16,6 +16,7 @@
         <project.reporting.outputEncoding>UTF-8</project.reporting.outputEncoding>
         <java.version>1.8</java.version>
         <jetty.version>${jetty94.version}</jetty.version>
+        <undertow.version>${undertow-legacy.version}</undertow.version>
 
         <springboot-version>2.4</springboot-version>
 

